modules/azure/nsg_test.go

```diff
@@ -0,0 +1,158 @@
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package azure
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2019-09-01/network"
+)
+
+func TestPortRangeParsing(t *testing.T) {
+	var cases = []struct {
+		portRange    string
+		expectedLo   int
+		expectedHi   int
+		expectsError bool
+	}{
+		{"22", 22, 22, false},
+		{"22-80", 22, 80, false},
+		{"*", 0, 65535, false},
+		{"*-*", 0, 0, true},
+		{"22-", 0, 0, true},
+		{"-80", 0, 0, true},
+		{"-", 0, 0, true},
+		{"80-22", 22, 80, false},
+	}
+
+	for _, tt := range cases {
+		t.Run(tt.portRange, func(t *testing.T) {
+			lo, hi, err := parsePortRangeString(tt.portRange)
+			if !tt.expectsError {
+				require.NoError(t, err)
+			}
+			assert.Equal(t, tt.expectedLo, int(lo))
+			assert.Equal(t, tt.expectedHi, int(hi))
+		})
+	}
+}
+
+func TestNsgRuleSummaryConverstion(t *testing.T) {
+	// Quick test to make sure the safe nil handling is working
+	var name = "test name"
+	var sdkStruct = network.SecurityRulePropertiesFormat{}
```

Comment:
NIT: Go idiom is to use `:=` when declaring and initializing:

```suggestion
	name := "test name"
	sdkStruct := network.SecurityRulePropertiesFormat{}
```

Reply:
Updated to use `:=` for initialization.